Editor/GitLogWindow.cs

```diff
@@ -49,6 +53,89 @@ public static async Task ShowFilesLog(IEnumerable<Module> modules, IEnumerable<s
+        public Texture2D avatar;
+        
+        public UserData(string email, string author, Color color)
+        {
+            this.email = email;
+            this.author = author;
+            this.color = color;
+        }
+    }
+
+    public static class GitUserBase
+    {
+        public const float AvatarSize = 96;
+
+        private static Dictionary<string, UserData> users = new();
+
+        public static UserData GetUserData(string email, string name = null)
+        {
+            if (users.ContainsKey(email))
+                return users[email];
+
+            if (name == null)
+                return null;
+
+            UnityEngine.Random.InitState(email.Length * name.Length);
+            var userData = new UserData(email, name, UnityEngine.Random.ColorHSV(0f, 0.6f, 0.6f, 0.8f, 0.6f, 0.8f));
+            users.Add(email, userData);
+            LoadAvatar(email);
+            return users[email];
+        }
+
+        private static async void LoadAvatar(string email)
+        {
+            string hashedEmail = await Task.Run(() => Md5Hash(email.Trim().ToLower()));
+            string avatarPath = Path.Combine(Application.temporaryCachePath, "mr-git-ui", $"{hashedEmail}.png");
+            
+            if (File.Exists(avatarPath))
+            {
+                var data = File.ReadAllBytes(avatarPath);
+                var cachedImage = new Texture2D(2, 2);
+                cachedImage.LoadImage(data);
+                users[email].avatar = cachedImage;
+                return;
+            }
+            
+            string url = $"https://www.gravatar.com/avatar/{hashedEmail}?s={AvatarSize}&d=retro";
+            var avatar = await DownloadTextureAsync(url);
+            avatar.name = hashedEmail;
+            users[email].avatar = avatar;
+            if (!Directory.Exists(Path.GetDirectoryName(avatarPath)))
+                Directory.CreateDirectory(Path.GetDirectoryName(avatarPath)!);
+            File.WriteAllBytes(avatarPath, avatar.EncodeToPNG());
+        }
+        
+        private static async Task<Texture2D> DownloadTextureAsync(string url)
+        {
+            using UnityWebRequest webRequest = UnityWebRequestTexture.GetTexture(url);
+            var asyncOp = webRequest.SendWebRequest();
+            
+            while (!asyncOp.isDone)
+                await Task.Delay(100);
+            
+            return webRequest.result == UnityWebRequest.Result.Success ? DownloadHandlerTexture.GetContent(webRequest) : null;
+        }
+
+        private static string Md5Hash(string input)
+        {
+            using MD5 md5 = MD5.Create();
+            byte[] inputBytes = Encoding.ASCII.GetBytes(input);
+            byte[] hashBytes = md5.ComputeHash(inputBytes);
+
+            var stringBuilder = new StringBuilder();
+            for (int i = 0; i < hashBytes.Length; i++) 
+                stringBuilder.Append(hashBytes[i].ToString("X2"));
+            return stringBuilder.ToString().ToLower();
```

Comment:
Looks like you can use `string.Join(null, hashBytes.Select(x => x.ToString("X2")))`
I think